iree/turbine/kernel/wave/compile.py

```diff
@@ -82,7 +83,10 @@ def wave_compile(options: WaveCompileOptions, kernel: "LaunchableWave") -> WaveK
     # Create an indexing context and populate substitutions.
     push(IndexingContext, IndexingContext())
     idxc = IndexingContext.current()
-    idxc.subs = options.subs
+
+    # Make a copy of the substitutions to avoid mutating the original
+    # options.subs.
+    idxc.subs = copy(options.subs)
```

Comment:
This fixes caching code because idxc subs get cached.  - Ivan

Reply:
`    idxc.subs = options.subs` is weird, but `copy(options.subs)` now it's strictly better.